source/feature-guide/feature-deriving.rst

```diff
@@ -624,6 +449,159 @@ ZCL function like:
 ...results in a private multiple\_response variable where the
 "Frequently" and "Always" categories are selected.
 
+
+Categorical Logic
+~~~~~~~~~~~~~~~~~
+
+Crunch provides logical operations like most languages do, but with a
+few enhancements.
+
+First, Crunch logic is actually based on *three* values so you can
+properly handle missing data. Many programming languages do this,
+too: SQL implements ``NULL`` in addition to ``TRUE`` and ``FALSE``,
+and general languages often have a similar ``None`` or ``NA`` value.
+
+In Crunch, rather than having a "boolean" type with true/false/null
+values, we instead allow *any* categorical variable to participate
+in logic. Categories which have the ``selected: true`` flag are
+"selected". Categories which have the ``missing: true`` flag are
+"missing", and any categories that lack both are "other".
+Any categorical variable that has at least one of each can function
+as a "logical variable".
```

Comment:
Maybe you should make it even more crystal clear that to be used as a logical variable, a categorical variable needs at least three categories: at least one selected category, at least one missing category, and at least one "other" category.

Reply:
I'm concerned that someone might read "at least one of each" as "at least one selected and one missing".

Reply:
That was actually wrong; fixed now.